### PR TITLE
i18n: German translations

### PR DESCRIPTION
**Pull request by mariokorte**
I've added most of the translations in German where localization support is already available.

### CODE REVIEW

**Comment by tiagobento** [comment]
Hi @mariokorte! Thanks for the contribution! @martinweiler will assist with the review of this PR.

I noticed that there are some places where the translations are not ready yet. Should we wait to review until you have everything translated? Let me know!

**Comment by mariokorte** [comment]
Hey @tiagobento and @martinweiler,

thanks for letting me know that there are still some rough edges. I will look into the missing parts and add them to this PR.

I will comment here when this is completed.

**Comment by mariokorte** [comment]
That completes my work on this for now.
If there are any problems or suggestions I'm happy to support further.

File: packages/boxed-expression-component/src/i18n/locales/de.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  builtInAggregator: "Integrierter Aggregator",
+  builtInAggregatorHelp: {
+    sum: "Gibt die Summe aller gesammelten Werte aus. Die Werte müssen numerisch sein.",
+    count: "Gibt die Anzahl passender Regeln aus.",
+    min: "Gibt den kleinsten Wert unter den Ergebnissen aus. Die resultierenden Werte müssen vergleichbar sein, z. B. Zahlen, Daten oder Text (lexikografische Reihenfolge).",
+    max: "Gibt den größten Wert unter den Ergebnissen aus. Die resultierenden Werte müssen vergleichbar sein, z. B. Zahlen, Daten oder Text (lexikografische Reihenfolge).",
+    none: "Aggregiert Werte in einer beliebigen Liste.",
+  },
+  choose: "Auswählen...",
+  class: "class",
```
Comment (martinweiler) [inline packages/boxed-expression-component/src/i18n/locales/de.ts]: Is this not translated on purpose?

File: packages/boxed-expression-component/src/i18n/locales/de.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  contextEntry: "KONTEXT-EINTRAG",
+  dataType: "Datentyp",
+  dataTypeDropDown: {
+    builtIn: "BUILT-IN",
+    custom: "CUSTOM",
+  },
+  decisionRule: "ENTSCHEIDUNGSREGEL",
+  decisionTable: "Entscheidungstabelle",
+  delete: "Löschen",
+  document: "document",
```
Comment (martinweiler) [inline packages/boxed-expression-component/src/i18n/locales/de.ts]: No translation

File: packages/online-editor/src/i18n/locales/de.ts [hunk 3]
```diff
@@ -0,0 +1,708 @@
...
+      title: (fileName: string) => `Editiere ${fileName}`,
+    },
+    alerts: {
+      setContentError: {
+        title: "Fehler beim Öffnen der Datei. Sie können sie als Text bearbeiten und das Diagramm nach der Korrektur erneut öffnen.",
+        action: "Als Text öffnen",
+      },
+      copy: "Content copied to clipboard",
+      updateGist: "Gist successfully updated.",
+      createGist: "Gist successfully created.",
```
Comment (martinweiler) [inline packages/online-editor/src/i18n/locales/de.ts]: Missing translation

File: packages/online-editor/src/i18n/locales/de.ts [hunk 4]
```diff
@@ -0,0 +1,708 @@
...
+      explanation: `${de_common.names.dmn}-Dateien werden zur Erstellung von Entscheidungsmodellen verwendet.`,
+      createNew: "Neues Entscheidungsmodell erstellen",
+    },
+    pmmlCard: {
+      title: `Scorecard-Modell (.${de_common.names.pmml})`,
+      explanation: `${de_common.names.pmml}-Dateien werden zur Erstellung von Scorecards verwendet`,
+      createNew: "Neue Scorecard erstellen",
+    },
+    trySample: "Beispiel ausprobieren",
+    chooseLocalFile: "Wähle eine lokale Datei",
```
Comment (martinweiler) [inline packages/online-editor/src/i18n/locales/de.ts]: Minor nitpicking - in other places the user is addressed formally, shouldn't it be the same here ("Wählen Sie eine lokale Datei")?

File: packages/online-editor/src/i18n/locales/de.ts [hunk 5]
```diff
@@ -0,0 +1,708 @@
...
+                " ",
+                wrapped("again"),
+                ` und wählen Sie dann "${de_common.terms.open}".`,
+              ],
+            },
+            alreadyRanBefore: `Wenn Sie die ${de_common.names.extendedServices} bereits installiert und ausgeführt haben:`,
+            launchExtendedServices: ["Starten Sie die ", wrapped("file")],
+            advanced: {
+              title: "Erweiterte Einstellungen",
+              runFollowingCommand: `Run the following command on a Terminal tab to start ${de_common.names.extendedServices} on a different port:`,
```
Comment (martinweiler) [inline packages/online-editor/src/i18n/locales/de.ts]: Missing translation

File: packages/online-editor/src/i18n/locales/de.ts [hunk 6]
```diff
@@ -0,0 +1,708 @@
...
+      organizations: "Bitbucket-Workspaces",
+    },
+    github: {
+      user: "GitHub-Benutzer",
+      organizations: "GitHub Organisationen",
+    },
+  },
+  gitStatusIndicatorActions: {
+    revert: {
+      title: "Revert",
```
Comment (martinweiler) [inline packages/online-editor/src/i18n/locales/de.ts]: Missing translation

File: packages/serverless-logic-web-tools/src/i18n/locales/de.ts [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+    },
+    alerts: {
+      setContentError: {
+        title: "Fehler beim Öffnen der Datei. Sie können sie als Text bearbeiten und das Diagramm nach der Korrektur erneut öffnen.",
+        action: "Als Text öffnen",
+      },
+      copy: "Content copied to clipboard",
+      updateGist: "Gist successfully updated.",
+      createGist: "Gist successfully created.",
+      errorPushingGist: "Push für das aktuelle Gist fehlgeschlagen. Push erzwungen?",
```
Comment (martinweiler) [inline packages/serverless-logic-web-tools/src/i18n/locales/de.ts:35]: Missing translation

File: packages/serverless-logic-web-tools/src/i18n/locales/de.ts [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+      title: (fileName: string) => `Editiere ${fileName}`,
+    },
+    alerts: {
+      setContentError: {
+        title: "Fehler beim Öffnen der Datei. Sie können sie als Text bearbeiten und das Diagramm nach der Korrektur erneut öffnen.",
+        action: "Als Text öffnen",
+      },
+      copy: "Content copied to clipboard",
+      updateGist: "Gist successfully updated.",
+      createGist: "Gist successfully created.",
```
Comment (martinweiler) [inline packages/serverless-logic-web-tools/src/i18n/locales/de.ts]: Missing translation

File: packages/serverless-logic-web-tools/src/i18n/locales/de.ts [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+                " ",
+                wrapped("again"),
+                ` und wählen Sie dann "${de_common.terms.open}".`,
+              ],
+            },
+            alreadyRanBefore: `Wenn Sie die ${de_common.names.extendedServices} bereits installiert und ausgeführt haben:`,
+            launchExtendedServices: ["Starten Sie die ", wrapped("file")],
+            advanced: {
+              title: "Erweiterte Einstellungen",
+              runFollowingCommand: `Run the following command on a Terminal tab to start ${de_common.names.extendedServices} on a different port:`,
```
Comment (martinweiler) [inline packages/serverless-logic-web-tools/src/i18n/locales/de.ts]: Missing translation

File: packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/resources/org/kie/workbench/common/stunner/client/widgets/resources/i18n/StunnerWidgetsConstants_de.properties [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+SessionPresenterView.Error=Fehler
+SessionPresenterView.Warning=Warnung
+SessionPresenterView.Info=Info
+SessionPresenterView.Notifications=Details über das Warnungs-Panel
+
+MarshallingResponsePopup.OkAction=Ok
+MarshallingResponsePopup.CancelAction=Abbrechen
+MarshallingResponsePopup.CopyToClipboardActionTitle=Kopieren der Nachrichten in die Zwischenablage
+MarshallingResponsePopup.LevelTableColumnName=Level
+MarshallingResponsePopup.MessageTableColumnName=Message
```
Comment (martinweiler) [inline packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/resources/org/kie/workbench/common/stunner/client/widgets/resources/i18n/StunnerWidgetsConstants_de.properties]: Missing translation

File: packages/serverless-workflow-diagram-editor/kie-wb-common-widgets/kie-wb-common-ui/src/main/resources/org/kie/workbench/common/widgets/client/resources/i18n/KieWorkbenchWidgetsConstants_de.properties [hunk 11]
```diff
@@ -1,7 +1,10 @@
 # translation auto-copied from project KIE Workbench - Common, version 6.0.0, document org.kie.workbench.widgets/kie-wb-common-ui/org/kie/workbench/common/widgets/client/resources/i18n/NewItemPopupConstants, author jdimanos
+KSessionSelectorViewImpl.KnowledgeBase=KIE Basis
```
Comment (martinweiler) [inline packages/serverless-workflow-diagram-editor/kie-wb-common-widgets/kie-wb-common-ui/src/main/resources/org/kie/workbench/common/widgets/client/resources/i18n/KieWorkbenchWidgetsConstants_de.properties]: Maybe leave it as "Kie Base"

File: packages/stunner-editors/kie-wb-common-widgets/kie-wb-common-ui/src/main/resources/org/kie/workbench/common/widgets/client/resources/i18n/KieWorkbenchWidgetsConstants_de.properties [hunk 12]
```diff
@@ -1,7 +1,10 @@
 # translation auto-copied from project KIE Workbench - Common, version 6.0.0, document org.kie.workbench.widgets/kie-wb-common-ui/org/kie/workbench/common/widgets/client/resources/i18n/NewItemPopupConstants, author jdimanos
+KSessionSelectorViewImpl.KnowledgeBase=KIE Basis
```
Comment (martinweiler) [inline packages/stunner-editors/kie-wb-common-widgets/kie-wb-common-ui/src/main/resources/org/kie/workbench/common/widgets/client/resources/i18n/KieWorkbenchWidgetsConstants_de.properties]: keep as "Kie Base"?

File: packages/stunner-editors/uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-client/src/main/resources/org/uberfire/ext/editor/commons/client/resources/i18n/Constants_de.properties [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+HtmlEditorView.OrderedList=Sortierte Liste
+HtmlEditorView.OutdentList=Liste ausrücken
+HtmlEditorView.IndentList=Liste einrücken
+HtmlEditorView.AlignLeft=Links ausrichten
+HtmlEditorView.AlignRight=Rechts ausrichten
+HtmlEditorView.AlignCenter=Zentriert ausrichten
+HtmlEditorView.Undo=Rückgängig
+HtmlEditorView.Redo=Wiederholen
+HtmlEditorView.SwitchToHtmlView=Zur HTML-Ansicht wechseln
+HtmlEditorView.SwitchView=SAnsicht wechseln
```
Comment (martinweiler) [inline packages/stunner-editors/uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-client/src/main/resources/org/uberfire/ext/editor/commons/client/resources/i18n/Constants_de.properties]: Typo

File: packages/serverless-workflow-diagram-editor/uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-client/src/main/resources/org/uberfire/ext/editor/commons/client/resources/i18n/Constants_de.properties [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+HtmlEditorView.OrderedList=Sortierte Liste
+HtmlEditorView.OutdentList=Liste ausrücken
+HtmlEditorView.IndentList=Liste einrücken
+HtmlEditorView.AlignLeft=Links ausrichten
+HtmlEditorView.AlignRight=Rechts ausrichten
+HtmlEditorView.AlignCenter=Zentriert ausrichten
+HtmlEditorView.Undo=Rückgängig
+HtmlEditorView.Redo=Wiederholen
+HtmlEditorView.SwitchToHtmlView=Zur HTML-Ansicht wechseln
+HtmlEditorView.SwitchView=SAnsicht wechseln
```
Comment (martinweiler) [inline packages/serverless-workflow-diagram-editor/uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-client/src/main/resources/org/uberfire/ext/editor/commons/client/resources/i18n/Constants_de.properties]: Typo

File: packages/form-dmn/src/i18n/locales/de.ts [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+          `Dieses ${de_common.names.dmn}-Modell enthält ein Konstrukt, das noch nicht unterstützt wird. Bitte beziehen Sie sich auf `,
+          wrapped("jira"),
+          " und melden Sie ein Problem. Vergessen Sie nicht, die aktuelle Datei hochzuladen.",
+        ],
+      },
+    },
+  },
+  validation: {
+    daysAndTimeError: "should match format P1D(ays)T2H(ours)3M(inutes)1S(econds)",
+    yearsAndMonthsError: "should match format P1Y(ears)2M(onths)",
```
Comment (martinweiler) [inline packages/form-dmn/src/i18n/locales/de.ts]: Missing translation

**Review by martinweiler — CHANGES_REQUESTED** [review]
Great translation @mariokorte - I just found some minor issues and a few missing translations.

**Comment by mariokorte** [comment]
@martinweiler thanks for your feedback. I will add the missing translations and apply your comments tomorrow.
There are one or two properties where I’m not 100% sure about the context, but if questions remain I will get back to you.

File: packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/resources/org/kie/workbench/common/stunner/client/widgets/resources/i18n/StunnerWidgetsConstants_de.properties (hunk 10 above)
Comment (mariokorte) [reply packages/serverless-workflow-diagram-editor/kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/resources/org/kie/workbench/common/stunner/client/widgets/resources/i18n/StunnerWidgetsConstants_de.properties]: I've translated 'Message' with 'Nachricht', but I would like to keep 'Level' as is intentionally. This is something a German understands fine and I'm not sure if 'Stufe' would be a better translation here.

**Comment by mariokorte** [comment]
@martinweiler I have incorporated your comments into the code. I would be happy if the requirements for the merge are met. If there are still wishes or further requirements I will be happy to help.

**Review by martinweiler — APPROVED** [review]
LGTM @mariokorte - thanks for your contribution and the excellent translation!

**Comment by mariokorte** [comment]
CI :: Code formatting run failed due to a syntax error (a superfluous bracket).
I have fixed the error, so hopefully the next run will succeed.

**Comment by mariokorte** [comment]
- fixed build error
- ran prettier to satisfy code format warnings

**Comment by mariokorte** [comment]
Some missing counteractions to the code formatting problems hid in another development branch of mine that needed to be included in this PR. Sorry.

Now it *really* should meet the code formatting requirements.

**Comment by mariokorte** [comment]
@tiagobento Checks have passed now and @martinweiler's remarks have been included. I guess this is ready for review to be merged.

**Review by yesamer — APPROVED** [review]
Thank you for your contribution, @mariokorte !

**Pull request merged**